docs/guide/error-handling.md

```
@@ -93,6 +93,112 @@ The `notFoundHandler()` middleware is placed before the error handler to catch u
 { "message": "Not Found" }
 ```
 
+## Framework errors with fix hints
+
+Framework-thrown errors (DI resolution failures, missing env vars, malformed module setup, etc.) use the `KickError` class — a structured error type with a stable `code`, a one-line `summary`, a `cause` explanation, an actionable `fix` block, and a `docsUrl`. The result is multi-line, scannable, and points at the exact change to apply:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)